tools/harness-thci/OpenThread_WpanCtl.py

```diff
@@ -297,23 +259,17 @@ def __sendCommand(self, cmd):
                     logging.info('%s: the read line is[%s]', self.port, line)
                     if line:
                         response.append(line)
-                        if re.match(WPAN_CARRIER_PROMPT, line):
+                        if re.match(self.prompt, line):
```

Comment:
nit: Is `self.prompt` a regular expression? If not, we should use `if line.startswith(self.prompt):`

Reply:
I am Ok with your suggestion. It is not really a re, but it supports part of the whole prompt like using "#" instead of "xxxx@xxx#".  @simonlingoogle @librasungirl @bukepo thoughts?

Reply:
I don't think it supports  part of the whole prompt like using "#" instead of "xxxx@xxx#".
```python
>>> re.match('#', 'xxxx@xxx#') is None
True
```
re.match() checks for a match only at the beginning of the string. so the current code is effective same as `line.startswith(self.prompt)` when there is no special chars in prompt.

Reply:
Yes, you are right, I mixed up with re.search.